File: dialoglibrary/src/main/java/com/sdg/dialoglibrary/test/ProgressDialog.java
package com.sdg.dialoglibrary.test;

import android.animation.ObjectAnimator;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.view.animation.Animation;
import android.view.animation.LinearInterpolator;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.sdg.dialoglibrary.R;
import com.sdg.dialoglibrary.SdgProgressBar;
import com.sdg.dialoglibrary.utils.SdgUtils;

public class ProgressDialog extends Dialog {

    private Context context;

    private CallBack callBack;

    //加载框的宽高（需保持一致，因此放一起）
    private int widthAndHeight = 140;
    private SdgProgressBar progressBar;

    public ProgressDialog(@NonNull Context context) {
        super(context);
        this.context = context;
    }

    public ProgressDialog(@NonNull Context context, int themeResId) {
        super(context, themeResId);
        this.context = context;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        View view = View.inflate(context, R.layout.dialog_progress,null);
        setContentView(view);

        Window win = getWindow();
        WindowManager.LayoutParams lp = win.getAttributes();
        lp.gravity = Gravity.CENTER;
//        lp.height = SdgUtils.dip2px(widthAndHeight);
//        lp.width = SdgUtils.dip2px(widthAndHeight);
        win.setAttributes(lp);

        progressBar = view.findViewById(R.id.progressBar);

        initData();

    }

    private void initData(){

    }

    public void setProgress(int progress){
        progressBar.setProgress(progress);
        if(progress >= 100){
            if(callBack != null) {
                callBack.success();
            }
        }
    }

    public ProgressDialog setWidthAndHeight(int widthAndHeight) {
        this.widthAndHeight = widthAndHeight;
        return this;
    }

    public ProgressDialog setCallBack(CallBack callBack) {
        this.callBack = callBack;
        return this;
    }

    public interface CallBack{
        void success();
    }
}

File: dialoglibrary/src/main/java/com/sdg/dialoglibrary/dialog/ProgressDialog.kt
package com.sdg.dialoglibrary.dialog

import android.os.Bundle
import android.view.View
import com.sdg.dialoglibrary.R
import com.sdg.dialoglibrary.utils.SdgUtils
import kotlinx.android.synthetic.main.dialog_progress.*

class ProgressDialog : BaseDialog() {

    //返回键取消
    private var backCancel: Boolean = false

    //弹框外部取消
    private var canceledOutside: Boolean = false

    //加载框的宽高（需保持一致，因此放一起）
    private var widthAndHeight:Float = 100f

    override fun getLayoutId(): Int = R.layout.dialog_progress

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
    }

    fun setProgress(progress:Int){
        progressBar.setProgress(progress)
    }

    fun setBackCancel(backCancel: Boolean): ProgressDialog {
        this.backCancel = backCancel
        return this
    }

    fun setOutsideCancel(canceledOutside: Boolean): ProgressDialog {
        this.canceledOutside = canceledOutside
        return this
    }

    fun setWidthAndHeight(widthAndHeight:Float):ProgressDialog{
        this.widthAndHeight = widthAndHeight
        return this
    }

    override fun windowWidth(): Int {
        return SdgUtils.dip2px(widthAndHeight)
    }

    override fun windowHeight(): Int {
        return SdgUtils.dip2px(widthAndHeight)
    }

}
